content/systems/filecoin_blockchain/struct/chain_manager/_index.md

```diff
@@ -13,32 +13,28 @@ The _Chain Manager_ is a central component in the blockchain system. It tracks a
 
 In so doing, the _chain manager_ is the central subsystem that handles bookkeeping for numerous other systems in a Filecoin node and exposes convenience methods for use by those systems, enabling systems to sample randomness from the chain for instance, or to see which block has been finalized most recently.
 
-The chain manager interfaces and functions are included here, but we expand on important details below for clarity.
 
-
-## Chain Expansion
+## Chain Extension
 
 ### Incoming block reception
 
-Once a block has been received and passes syntactic and semantic validation it must be added to the local datastore, regardless whether it is understood as the best tip at this point. Future blocks from other miners may be mined on top of it and in that case we will want to have it around to avoid refetching.
+Once a block has been received and passes syntactic and semantic validation (as explained above) it must be added to the local datastore, regardless of whether it is understood as the best tip at this point. Future blocks from other miners may be mined on top of it and in that case we will want to have it around to avoid refetching.
```

Comment:
```suggestion

```

This entire line sounds like implementation details. We can rework it if we think it's sound advise and worth staying here but the term *must* should be used with care in a spec I think.

Reply:
I'm fine with just changing the word if you think it's useful. It still seems to me too much of an implementation detail, should we really store those tipsets preemptively? Is it really worth the disk space to save some bandwidth? Maybe yes, maybe no, but you see we are already talking about details that don't reflect how the protocol works. They sound to me more of an engineering issue than anything else.